src/interactions/panZoomInteraction.ts

```diff
@@ -8,8 +8,8 @@ export module Interactions {
      */
     private static _PIXELS_PER_LINE = 120;
 
-    private _xScale: QuantitativeScale<any>;
-    private _yScale: QuantitativeScale<any>;
+    private _xScales: QuantitativeScale<any>[];
+    private _yScales: QuantitativeScale<any>[];
```

Comment:
Yeah. It pushes the weirdness into the setter/getters, but avoids duplicating the `Set` logic. It also reduces the chance of array-mutation bugs when users operate on the return values of `xScales()`/`yScales()`/=.